agent/bench-scripts/pbench-fio

```diff
@@ -916,14 +916,23 @@ function fio_run_benchmark() {
 					chmod +x "$iteration_dir/process-iteration-samples.cmd"
 					$iteration_dir/process-iteration-samples.cmd
 					fail=$?
-					if [ $fail -ne  0 ]; then
+					if [ $fail -eq 0 ]; then
+						# This sample was successful, so break out of the
+						# retry loop.
+						repeat_iteration=0
+					else
+						# The sample was not within the max stddev.
 						((failures++))
-					fi
-					if [ $failures -ge $max_failures ]; then
-						((iteration_failures++))
-					fi
-					if [ $fail -eq 0 -o $failures -ge $max_failures ]; then
-						break 1
+						if [ $failures -ge $max_failures ]; then
+							# We consider this iteration to be a failure since too
+							# many sample failures were encountered.
+							((iteration_failures++))
+							repeat_iteration=0
+						fi
+						if [ "$postprocess_only" = "n" ]; then
+							# Do not repeat the iteration in post-processing only mode.
+							repeat_iteration=0
+						fi
 					fi
```

Comment:
If I might suggest:
```
					if [ $fail -eq 0 -o "$postprocess_only" = "n" ]; then
						# This sample was successful, 
						# or we're in post-processing
						# only mode, so break out of
						# the retry loop.
						break
					fi
					# The sample was not within the max stddev.
					((failures++))
					if [ $failures -ge $max_failures ]; then
						# We consider this iteration to
						# be a failure since too many
						# sample failures were encountered.
						((iteration_failures++))
						break
					fi
```
This would allow us to get rid of the `repeat_iteration` variable by changing line 821 to an "infinite" loop.  (This approach won't count any failures during postprocess-only operation, but I don't think that changes overall behavior.)

Reply:
Ick, I really try to avoid explicit infinite loop constructs, but okay.

Reply:
I figure, if you buy into using `break` statements, then all you've really done is moved the exit test from the top of the loop to a more convenient point...so, it's not actually an "infinite loop", it's a loop with a "deferred test".  😁

Reply:
Oh, and, when one gets to the point where s/he is creating control variables just to make the loop exit test work right, it's probably time to switch to a "deferred test" loop.  😉